src/biokbase/narrative/jobs/jobmanager.py

```diff
@@ -79,39 +76,35 @@ def initialize_jobs(self):
         for job_id, job_state in job_states.items():
             job = Job.from_state(job_state)
             status = job_state.get("status")
-            refresh_state = 1 if status not in TERMINAL_STATES else 0
+            refresh_state = 1 if status not in TERMINAL_STATUSES else 0
             self._running_jobs[job_id] = {
                 "refresh": refresh_state,
                 "job": job,
             }
-            # if the job is in a terminal state, add it to the _completed_job_states index
-            if not refresh_state:
-                revised_state = self._construct_job_status(
-                    self.get_job(job_id), job_state
-                )
-                self._completed_job_states[job_id] = revised_state
 
-    def _create_jobs(self, job_ids):
+    def _create_jobs(self, job_ids, job_states=None):
         """
         TODO: error handling
         Given a list of job IDs, creates job objects for them and populates the _running_jobs dictionary
         """
-        job_states = clients.get("execution_engine2").check_jobs(
-            {
-                "job_ids": job_ids,
-                "return_list": 0,
-                "exclude_fields": JOB_INIT_EXCLUDED_JOB_STATE_FIELDS,
-            }
-        )
+        job_ids = [job_id for job_id in job_ids if job_id not in self._running_jobs]
+        if not job_states:
+            job_states = clients.get("execution_engine2").check_jobs(
+                {
+                    "job_ids": job_ids,
+                    "exclude_fields": JOB_INIT_EXCLUDED_JOB_STATE_FIELDS,
+                }
+            )
         for job_id in job_ids:
-            if job_id in job_ids and job_id not in self._running_jobs:
-                job_state = job_states.get(job_id, {})
-                job = Job.from_state(job_state)
-                # Note that when jobs for this narrative are initially loaded,
-                # they are set to not be refreshed. Rather, if a client requests
-                # updates via the start_job_update message, the refresh flag will
-                # be set to True.
-                self._running_jobs[job_id] = {"refresh": 0, "job": job}
+            job_state = job_states.get(job_id, {})
+            job = Job.from_state(job_state)
+            # Note that when jobs for this narrative are initially loaded,
+            # they are set to not be refreshed. Rather, if a client requests
+            # updates via the start_job_update message, the refresh flag will
+            # be set to True.
+            self._running_jobs[job_id] = {"refresh": 0, "job": job}
```

Comment:
Can this code be combined with the code in `initialise_jobs` so that we are treating new jobs consistently? Is there anywhere else where new jobs are created that this should be dealt with?

Reply:
Ha! I should read your notes before saying the same thing elsewhere... :)

Reply:
We decided that this can wait because the logic is slightly different (how `"refresh"` is initialized, the ee2 method). But there is also `register_new_job` which can be refactored in, and lines can be condensed so it looks less aesthetically like duplicate logic